### DISCUSSION TITLE
Add Rack Group column in GUI for all Devices view

### DISCUSSION THREAD

**Discussion by amat3ur85** [netbox-community/netbox#5993] [Ideas]
It would be beneficial to have possibility to select Rack group as "column" in netbox.test.com/dcim/devices/ view.

Use case:
Let us say that we have multiple rows on the same site (row 55 and row60), and each row (rack group) has racks from 1 to 120.
Based on current all devices view we won't be able to distinguish where, in which rack_group/rack combination is specific device racked.

Currently we have:

![rack_group](https://user-images.githubusercontent.com/53473669/111328204-537a9880-866e-11eb-9f1e-ca6bf4b2ed18.png)

Addition:

![rack_group_added](https://user-images.githubusercontent.com/53473669/111328220-583f4c80-866e-11eb-85d1-ef253986e6c4.png)

**Comment by xkilian** [comment]
Just to clarify, I believe you would like the first container (location/rack group) a rack is assigned to.  (For some users it would be be a row for others it would be the name of the room.

**Reply by amat3ur85** [reply]
That is correct, it is dcim/rack-groups, but we all have different understanding and usage of those groupings. That data is already present in full scope on the specific device, so it would be beneficial to have it in all devices view.


![rack_group_menu](https://user-images.githubusercontent.com/53473669/111340192-b3763c80-8678-11eb-8c6b-7fa53c32901d.png)


In previous example:
![rack_groups_list](https://user-images.githubusercontent.com/53473669/111338686-69408b80-8677-11eb-90dd-62d27c100361.png)

So either of 2 representation would be good.
- To have two columns. One for rack as is + different column named rack_group
- To have rack column changed (that is currently there) to represent {rack_group}/{rack}, in the same way it was already present on specific node. 

![rack_groups_representation](https://user-images.githubusercontent.com/53473669/111340278-c721a300-8678-11eb-979f-67bb1f7d6bb8.png)

**Comment by candlerb** [comment]
Soon it will be possible to assign a device directly to a rackgroup without assigning it to a rack (#4971, #5895), in which case I think this will fall out automatically.

The device itself will gain a rackgroup (location) attribute, so it won't be necessary to lookup `device > rack > rackgroup`

**Reply by amat3ur85** [reply]
Not sure how this excludes any of those.
Currently if you list all devices, there is just no data about rack-group, future location in columns optimization of shown table.
You cannot select that column now. If future changes will add this column as well, that will cover this scenario.

![column_change](https://user-images.githubusercontent.com/53473669/111346845-e8858d80-867e-11eb-8688-51e5d6e5afcc.png)

**Reply by candlerb** [reply]
That's right: currently a device has a "rack" attribute, but not a "rackgroup" attribute (because "rackgroup" is an attribute of the rack, not the device).

I understand this is going to change, so that a device can be associated directly with a rackgroup.

**Reply by amat3ur85** [reply]
That is cool! Thank you for the info.
Didn't know that something like that is comming.
👍
